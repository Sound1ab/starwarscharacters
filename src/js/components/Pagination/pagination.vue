<template>
	<div class="pagination">
		<div
			v-if="previous"
			data-test="previous"
			class="pagination__chevron"
			@click="handlePrevious"
		>
			<svgicon
				name="chevron-left"
				:color="'white'"
				width="50"
				height="50"
			>
			</svgicon>
		</div>
		<span data-test="current-page">
			{{currentPage}}
		</span>
		<div
			v-if="next"
			data-test="next"
			class="pagination__chevron"
			@click="handleNext"
		>
			<svgicon
				name="chevron-right"
				:color="'white'"
				width="50"
				height="50"
			>
			</svgicon>
		</div>
	</div>
</template>

<script>
	import '@/assets/compiled-icons/chevron-left';
	import '@/assets/compiled-icons/chevron-right';
	import {mapState} from 'vuex';
	export default {
		name: 'pagination',
		computed: {
			// Previous and next returned from api response
			...mapState({
				previous: state => state.fetch.previous,
				next: state => state.fetch.next
			}),
			currentPage () {
				return parseInt(this.$route.params.id, 10);
			}
		},
		methods: {
			handlePrevious () {
				this.$router.push({name: 'page', params: {id: this.currentPage - 1}});
			},
			handleNext () {
				this.$router.push({name: 'page', params: {id: this.currentPage + 1}});
			}
		}
	};
</script>

<style lang="scss" type="text/scss">
	.pagination {
		padding: em(40);
		display: flex;
		justify-content: center;
		align-items: center;
		&__chevron {
			cursor: pointer;
		}
	}
</style>
