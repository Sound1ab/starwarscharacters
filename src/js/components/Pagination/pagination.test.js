/* eslint-disable */
import {shallow, createLocalVue} from '@vue/test-utils';
import {createRenderer} from 'vue-server-renderer';
import Vuex from 'vuex';
import * as svgicon from 'vue-svgicon';
import Pagination from './pagination';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(svgicon);

let mountComponent = (store, stub) => shallow(Pagination, {
	localVue,
	store,
	mocks: {
		$route: {
			params: {
				id: 2
			}
		},
		$router: {
			push: stub
		}
	}
});

describe('Pagination', () => {
	let wrapper;
	let store;
	let stub;
	beforeEach(() => {
		stub = jest.fn();
		store = new Vuex.Store({
			state: {
				fetch: {
					previous: true,
					next: true
				}
			}
		});
		wrapper = mountComponent(store, stub);
		jest.clearAllMocks();
	});
	afterEach(() => {
		wrapper.destroy();
	});
	describe('htmlStructure', () => {
		it('html snapshot matches', () => {
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
	describe('component', () => {
		it('vm - is instantiated', () => {
			expect(wrapper.isVueInstance()).toBe(true);
		});
	});
	describe('@events', () => {
		it('should push previous route when left chevron is clicked', () => {
			let previous = wrapper.find('[data-test="previous"]');
			previous.trigger('click');
			expect(wrapper.vm.$router.push).toBeCalledWith({name: 'page', params: {id: 1}});
		});
		it('should push next route when right chevron is click', () => {
			let next = wrapper.find('[data-test="next"]');
			next.trigger('click');
			expect(wrapper.vm.$router.push).toBeCalledWith({name: 'page', params: {id: 3}});
		});
	});
	describe('interface', () => {
		it('should show the current page number', () => {
			let currentPage = wrapper.find('[data-test="current-page"]');
			expect(currentPage.text()).toBe('2');
		});
		it('should hide left chevron when on first page', () => {
			expect(wrapper.find('[data-test="previous"]').exists()).toBe(true);
			wrapper.vm.$store.state.fetch.previous = false;
			expect(wrapper.find('[data-test="previous"]').exists()).toBe(false);
		});
		it('should hide right chevron when on last page', () => {
			expect(wrapper.find('[data-test="next"]').exists()).toBe(true);
			wrapper.vm.$store.state.fetch.next = false;
			expect(wrapper.find('[data-test="next"]').exists()).toBe(false);
		});
	});
});
