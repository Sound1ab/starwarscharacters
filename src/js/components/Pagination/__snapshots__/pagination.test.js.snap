// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination htmlStructure html snapshot matches 1`] = `
<div class="pagination">
    <div data-test="previous" class="pagination__chevron">
        <!---->
    </div> <span data-test="current-page">
		2
	</span>
    <div data-test="next" class="pagination__chevron">
        <!---->
    </div>
</div>
`;
